validation-test/stdlib/FixedPointConversion.swift.gyb

```diff
@@ -0,0 +1,159 @@
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../utils/gyb %s -o %t/FixedPointConversion.swift
+// RUN: %S/../../utils/line-directive %t/FixedPointConversion.swift -- %target-build-swift %t/FixedPointConversion.swift -o %t/a.out_Debug
+// RUN: %S/../../utils/line-directive %t/FixedPointConversion.swift -- %target-build-swift %t/FixedPointConversion.swift -o %t/a.out_Release -O
+//
+// RUN: %S/../../utils/line-directive %t/FixedPointConversion.swift -- %target-run %t/a.out_Debug
+// RUN: %S/../../utils/line-directive %t/FixedPointConversion.swift -- %target-run %t/a.out_Release
+// REQUIRES: executable_test
+
+%{
+
+from SwiftIntTypes import all_integer_types
+
+word_bits = 4
+
+def intMax(bits, signed): 
+    bits = bits - 1 if signed else bits 
+    return (1 << bits) - 1
+    
+def intMin(bits, signed):
+  return -1 * intMax(bits, signed) - 1 if signed else 0
+
+floatNameToSignificandBits = { 'Float32':24, 'Float64':53, 'Float80':64 }
+  
+}%
+
+import StdlibUnittest
+
+var FixedPointConversionTraps = TestSuite("FixedPointToFixedPointConversionTraps")
+var FixedPointConversionFailure = TestSuite("FixedPointToFixedPointConversionFailures")
+
+var FloatingPointConversionTruncations = TestSuite("FloatingPointToFixedPointConversionTruncations")
+var FloatingPointConversionFailure = TestSuite("FloatingPointToFixedPointConversionFailures")
+
+% for self_ty in all_integer_types(word_bits):
+%   selfBits = self_ty.bits
+%   selfSigned = self_ty.is_signed
+%   selfMin = intMin(selfBits, selfSigned)
+%   selfMax = intMax(selfBits, selfSigned)
+%   Self = self_ty.stdlib_name
+
+%   for other_ty in all_integer_types(word_bits):
+%     otherBits = other_ty.bits
+%     otherSigned = other_ty.is_signed
+%     otherMin = intMin(otherBits, otherSigned)
+%     otherMax = intMax(otherBits, otherSigned)
+%     Other = other_ty.stdlib_name
+
+%     for testValue in [selfMin, selfMax, selfMin - 1, selfMax + 1, otherMin, otherMax]:
+
+%       if testValue < otherMin or testValue > otherMax: 
+%          # Do nothing/continue.
+
+%       elif testValue >= selfMin and testValue <= selfMax:
+
+/// Always-safe conversion from ${Other}(${testValue}) to ${Self}.
+FixedPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}") {
+  // Test that nothing interesting happens and we end up with the same result after converting.
+  let input = get${Other}(${testValue})
+  let result = ${Self}(input)
+  expectEqual(${testValue}, result)
+  _blackHole(result)
+}
+
+/// Never-nil failable conversion from ${Other}(${testValue}) to ${Self}.
+FixedPointConversionFailure.test("${Other}To${Self}FailableConversion/dest=${testValue}") {
+  // Test that nothing interesting happens and we end up with a non-nil, identical result.
+  let input = get${Other}(${testValue})
+  var result = ${Self}(exactly: input)
```

Comment:
Here both `Self` and `Other` are integer types, and there are no initializers with `exact` argument label on integers that accept other integers.